src/lib.rs

```diff
@@ -383,7 +385,7 @@ impl Config {
         if let Some(ref generator) = self.generator {
             is_ninja = generator.to_string_lossy().contains("Ninja");
         }
-        if target.contains("windows-gnu") {
+        if target_triplet.contains("windows-gnu") {
```

Comment:
I think that would be a good direction for this codebase to move towards. I don't have the bandwidth to refactor it though. My thinking is basically that by providing the base structure for organization of targets, it can slowly move in that direction. However, let me know if you'd prefer I get rid of `Target` and inline everything instead.